---
layout: default
---

<a name="top"></a>

# Week 8 lecture and tutorial
1. [The genome and the transcriptome](#transcriptome)
2. [Layers of the transcriptome](#layers)
3. [Gene expression](#gene)
4. [Tools for studying gene expression](#study)
	- **A.** [microarrays](#micro)
	- **B.** [RNAseq](#RNAseq)
5. [RNAseq workflow](#workflow)
	- **A.** [Library prep](#library)
	- **B.** [In silico analysis](#silico)
6. [Downstream analyses of RNAseq data](#down)
	- **A.** [Differential expression](#diff)
	- **B.** [Gene co-expression networks](#net)
7. [Tutorial assignment](#tut)


## <ins>**The genome and the transcriptome**</ins> <a name="transcriptome"></a>
- Gen**ome** = all the genes in a cell/tissue/organism
- Transcrip**ome** = all the transcripts (RNAs) in a cell/tissue/organism
![trans01](/Images/Week08/trans01.png)

The transcriptome is highly dynamic. The ability to <ins>profile</ins> the transcriptome gives us extremely important clues about the cellular function of each gene.

## <ins>**Layers of the transcriptome**</ins> <a name="layers"></a>
The full transcriptome includes all types of RNAs present in a cell. However, in many cases, the term 'transcriptome' tends to refer to the full set of mRNAs (protein-coding transcripts) in the cell.

![trans02](/Images/Week08/trans02.png)

## <ins>**Gene expression**</ins> <a name="gene"></a>

The transcriptome essentially gives us a readout of gene expression for all the genes in a genome in a given cell/tissue/organism. 

- The transcriptome differs across developmental stages. A different set of gene is expressed in the embryo than in an adult.
	- Identifying which genes are expressed when gives us a headstart on identifying genes involved in development.

![trans03](/Images/Week08/trans03.png)

Observing differences in the transcriptome is akin to observing differences in gene regulation.

![trans04](/Images/Week08/trans04.png)

Gene expression also differs between cell types and tissues in an organism.
- The image below is a readout from a plant genome/transcriptome database called [The Arabidopsis Information Resource (TAIR)](https://www.arabidopsis.org/)
	- Try searching on gene on TAIR:
 		- Search `TIC110` (or any gene of interest)
   		- Scroll down to `Expression`
		- Set Data Source to `Developmental_map` 

![trans05](/Images/Week08/trans05.png)

![trans06](/Images/Week08/trans06.png)

## <ins>**Tools for studying gene expression**</ins> <a name="study"></a>

![trans10](/Images/Week08/trans10.png)

### <ins>**microarrays**</ins> <a name="micro"></a>

![trans07](/Images/Week08/trans07.png)

![trans08](/Images/Week08/trans08.png)

![trans09](/Images/Week08/trans09.png)

![trans11](/Images/Week08/trans11.png)

### <ins>**RNAseq**</ins> <a name="RNAseq"></a>

![trans12](/Images/Week08/trans12.png)

## <ins>**RNAseq workflow**</ins> <a name="workflow"></a>

![trans13](/Images/Week08/trans13.png)

### <ins>**Library prep**</ins> <a n	ame="library"></a>

![trans14](/Images/Week08/trans14.png)

### <ins>**In silico analysis**</ins> <a name="silico"></a>

![trans15](/Images/Week08/trans15.png)

## <ins>**Downstream analyses of RNAseq data**</ins> <a name="down"></a>

![trans18](/Images/Week08/trans18.png)

![trans19](/Images/Week08/trans19.png)

### <ins>**Differential expression**</ins> <a name="diff"></a>

![trans16](/Images/Week08/trans16.png)

### <ins>**Gene co-expression networks**</ins> <a name="net"></a>

![trans17](/Images/Week08/trans17.png)

## <ins>**Tutorial assignment**</ins> <a name="tut"></a>



[Back to Top](#top)
